sycl/doc/extensions/experimental/sycl_ext_oneapi_cuda_async_barrier.asciidoc

```diff
@@ -0,0 +1,386 @@
+= sycl_ext_oneapi_cuda_async_barrier
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+[NOTE]
+====
+This extension is currently implemented in {dpcpp} only for NVIDIA GPU devices 
+with Compute Capability of 8.0 or above and
+only when using the CUDA backend.
+
+====
+
+== Introduction
+
+This document describes an extension that adds `barrier`, which acts similarly 
+to https://en.cppreference.com/w/cpp/thread/barrier[`std::barrier`].
+
+== Specification
+
+=== Feature Test Macro
```

Comment:
The latest extension [template](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc#feature-test-macro) has different text for experimental extensions.